### DISCUSSION TITLE
Thousands of folders are created and affects tags and performance

### DISCUSSION THREAD

**Discussion by juanpeich** [photoprism/photoprism#3637] [Q&A]
Hello family, 

Thank you so much for your great work. 

I have photoprism on a raspberry pi 3B+ The problem is that it creates too many thumbnails for me (right now I only have about 20 original photos). 
I have tried to delete the "sidecar" folder, the "storage/cache/thumbnails" folder. I have disabled backups and have followed the configuration indicated in the documentation. But I always get too many thumbnails and I think that is slowing down photoprism a lot. 

The problem is that even in the labels, it recognizes the thousands of thumbnails as images and catalogs them for me. If I have about 30 original photos, how can it be that I have more than 3000 folders and I don't know how many more images have been created (All images are the same? 
Can you tell me some more optimal configuration and how to restore photoprism series to start clean without thousands of folders, labels, etc? 

I am attaching some screenshots so that you can see the folders that are generated with 20 photos and also all the images that I am supposed to have in the different labels (thousands of images being the same thumbnail):

![Screenshot_2023-08-16-00-06-16-030_com android chrome](https://github.com/photoprism/photoprism/assets/108826500/634ca301-5413-4f96-9d9a-82f9f560e69b)
![Screenshot_2023-08-16-00-06-46-906_com android chrome](https://github.com/photoprism/photoprism/assets/108826500/16fdea7f-2c08-41e6-a3e8-c36464960e79)
![Screenshot_2023-08-16-00-07-05-819_com android chrome](https://github.com/photoprism/photoprism/assets/108826500/dcec3e2f-65d3-4214-a955-43cd80c55023)

The other question is that if I have to change equipment, since I want to connect a 4TB hdd to it, would a raspberry pi 4 8Gb be enough or do I need to buy a miniPc?

Thank you very much for everything beforehand. JuamPe

**Comment by graciousgrey** [comment]
Is is possible that your storage folder is configured to be within you originals folder? This would explain why that many thumbnails are generated.

See also:
- https://docs.photoprism.app/getting-started/docker-compose/#photoprismstorage

**Reply by juanpeich** [reply]
I pass you the configuration of docker-compose:

```
`` volumes:
      - "/mnt/discoUSB/storage:/photoprism/storage"     # Cache and sidecar storage folder (WRITABLE)
      - "/mnt/discoUSB/originals:/photoprism/originals" # Original media folder (DO NOT REMOVE)
      - "/mnt:/photoprism/originals/mnt:slave"            # Mounted media folders (optional)
      - "/mnt/discoUSB/import:/photoprism/import"       # Import source folder (optional)
```

I have added an HDD where the photos will be stored, and that is why the path is changed.

I have tried putting what you tell me, but if I do this, it does not enter the photoprism web interface:

```
`` volumes:
      - "./storage:/photoprism/storage" # "/mnt/discoUSB/storage:/photoprism/storage"     # Cache and sidecar storage folder (WRITABLE)
      - "/mnt/discoUSB/originals:/photoprism/originals" # Original media folder (DO NOT REMOVE)
      - "/mnt:/photoprism/originals/mnt:slave"            # Mounted media folders (optional)
      - "/mnt/discoUSB/import:/photoprism/import"       # Import source folder (optional)
```

Best regards,
JuamPe

**Comment by lastzero** [comment]
👉 https://docs.photoprism.app/known-issues/#nested-storage-folder

**Comment by juanpeich** [comment]
Hi again,

Could someone give me a hand please?


https://github.com/photoprism/photoprism/discussions/3637#discussioncomment-6763546

Thank you very much in advance
JuamPe

**Reply by lastzero** [reply]
`mnt:/photoprism/originals/mnt:slave` seems wrong in your config. Please note that our team is currently out of the office, so we can only provide limited support.

**Reply by juanpeich** [reply]
@lastzero The compose settings are default except for the line I changed. It is the one that came with the photoprism image for raspberry. What would be the default configuration for comparison? I've read this thread, but I'm bad at programming and I don't know exactly what I should change: https://github.com/photoprism/photoprism-docs/blob/master/docs/getting-started/docker-compose.md Thanks in advance and greetings JuamPe

**Reply by lastzero** [reply]
You've mounted /mnt into originals, so the import folder for copying files to originals is in originals. That's a loop.

**Reply by lastzero** [reply]
https://docs.photoprism.app/known-issues/#nested-import-folder

**Reply by juanpeich** [reply]
Returning to the topic, I can't get the images to stay on the external HDD of the raspberry pi4.
I have tried changing the paths within docker-compose.yml, but to no avail. 

How should I do it if I have mounted the HDD from /dev/sda in /media/HDD?

Thank you very much in advance and forgive me for my ignorance.
